app/controllers/spotlight/resources/csv_upload_controller.rb

```diff
@@ -14,7 +14,7 @@ class CsvUploadController < ApplicationController
 
       def create
         file = csv_params[:url]
-        csv = CSV.parse(file.read, headers: true, return_headers: false, encoding: 'utf-8').map(&:to_hash)
+        csv = CSV.parse(File.read(file.tempfile.to_path.to_s, encoding: 'UTF-8'), headers: true, return_headers: false, encoding: 'utf-8').map(&:to_hash)
```

Comment:
I am sure that there will be a tempfile.  My reasoning is that the variable "file" refers to an instance of ActionDispatch::Http::UploadedFile, not an instance of IO.  I can see this in the debugger.  UploadedFile by definition always has a tempfile according to http://api.rubyonrails.org/classes/ActionDispatch/Http/UploadedFile.html since its purpose is to automatically close and dispose of the file when the read method is called.

If you would prefer, here's another way we could fix the bug:

        csv = CSV.parse(file.read.force_encoding('UTF-8'), headers: true, return_headers: false, encoding: 'utf-8').map(&:to_hash)

Is there any reason to prefer one solution over the other?